File: customGTApp/src/main/java/com/customGTApp/service/ServiceProdService.java
package com.customGTApp.service;

import com.customGTApp.model.Photo;
import com.customGTApp.model.ServiceProd;

import java.util.List;
/**
 * ServiceProdService interface that will be implemented by ServiceProdServiceImpl and other classes,
 * making it easier to manage the ServiceProd model. Also, making the code more readable and easier to maintain.
 */
public interface ServiceProdService {
    /**
     * Method to return all services
     * @return a list of all services
     */
    List<ServiceProd> getAllServices();
    /**
     * Method to find a service by id
     * @param id the id of the service we want to find
     * @return the service found
     */
    ServiceProd getServiceById(Long id);
    /**
     * Method to add a service
     * @param serviceProd the service that will be added
     * @return the service that was added
     */
    ServiceProd addService(ServiceProd serviceProd);
    /**
     * Method to update a service
     * @param serviceProd the service that will be updated
     * @return the service that was updated
     */
    ServiceProd updateService(ServiceProd serviceProd);
    /**
     * Method to delete a service by id
     * @param id the id of the service that will be deleted
     */
    void deleteServiceById(Long id);

    /**
     * Method to get all the photos of a service
     * @param serviceId the id of the service we want to get the photos
     * @return the photos list of the service
     */
    List<Photo> getAllServicePhotos(Long serviceId);

    /**
     * Method to update the price of a service
     * @param serviceId the id of the service we want to update the price
     * @param price the new price of the service
     * @return the service with the updated price
     */
    ServiceProd updatePrice(Long serviceId, float price);
}

File: customGTAppFront/src/app/services/service-prod.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs/internal/Observable';
import { ServiceProd } from '../models/serviceProd';
import { Photo } from '../models/photo';

@Injectable({
  providedIn: 'root'
})
export class ServiceProdService {
  private apiUrl = environment.apiUrl + '/service';

  constructor(private http: HttpClient) { }

  public getAllServices(): Observable<ServiceProd[]> {
    return this.http.get<ServiceProd[]>(this.apiUrl + '/all');
  }

  public getServiceById(id: number): Observable<ServiceProd> {
    return this.http.get<ServiceProd>(this.apiUrl + '/find/' + id);
  }

  public addService(service: ServiceProd): Observable<ServiceProd> {
    return this.http.post<ServiceProd>(this.apiUrl + '/add', service);
  }

  public updateService(service: ServiceProd): Observable<ServiceProd> {
    return this.http.put<ServiceProd>(this.apiUrl + '/update', service);
  }

  public deleteServiceById(id: number): Observable<ServiceProd> {
    return this.http.delete<ServiceProd>(this.apiUrl + '/delete/' + id);
  }

  public getAllPhotosByServiceId(serviceId: number): Observable<Photo[]> {
    return this.http.get<Photo[]>(this.apiUrl + '/' + serviceId + '/photos');
  }

  public updatePrice(serviceId: number, price: number): Observable<ServiceProd> {
    const url = `${this.apiUrl}/updatePrice/${serviceId}?price=${price}`;
    return this.http.put<ServiceProd>(url, null);
  }

}
